### PR TITLE
Remove the release submodule

### PR DESCRIPTION
**Pull request by berendt**
Currently, the release submodule is used as a source for versions as well as various curl calls on Github to the release repository.

It is sufficient to clone the release repository at build time and use only this clone as the source for the versions.

Signed-off-by: Christian Berendt <berendt@osism.tech>

### CODE REVIEW

File: scripts/001-prepare.sh [hunk 1]
```diff
@@ -20,25 +20,24 @@ PROJECT_REPOSITORY=https://github.com/openstack/kolla
...
-fi
-
 . defaults/all.sh
 . defaults/$OPENSTACK_VERSION.sh
 
 export VERSION
 export OPENSTACK_VERSION
 
-git submodule update --remote
+git clone https://github.com/osism/release
```
Comment (osfrickler) [inline scripts/001-prepare.sh]: Can we make this optional, to be executed only if the target directory does not exist yet? In the Zuul CI, the repo would already be present via some `required-projects` dependency.

We also then need either the repo path to be configurable or possibly create a symlink. But IMO making on-the-fly changes within the working repo should be done as seldom as possible, so being able to point to a path outside would be my preferred approach.

File: scripts/001-prepare.sh [hunk 2]
```diff
@@ -20,25 +20,24 @@ PROJECT_REPOSITORY=https://github.com/openstack/kolla
...
 
 if [[ ! -e release/$VERSION/base.yml ]]; then
     echo "release $VERSION does not exist"
     exit 1
 fi
 
+# NOTE: For builds for a specific release, the OpenStack version is taken from the release repository.
+if [[ $VERSION != "latest" ]]; then
+    OPENSTACK_VERSION=$(grep "openstack_version:" release/$VERSION/openstack.yml | awk -F': ' '{ print $2 }')
```
Comment (osfrickler) [inline scripts/001-prepare.sh:44]: Nit: If you use ': ' as field separator, you should also search for 'openstack_version: '.

Also the grep is redundant, you can add /openstack_version:/ to the awk statement instead.

Comment (berendt) [reply scripts/001-prepare.sh:44]: I'll do that in a follow-up.

File: scripts/001-prepare.sh (hunk 1 above)
Comment (berendt) [reply scripts/001-prepare.sh]: 1. part: done
2. part: I'll do that in a follow-up. The kolla repository is currently also being cloned directly. We can then adapt both.

**Review by osfrickler — APPROVED** [review]

**Pull request merged**